core/client/fs/src/main/java/alluxio/client/file/UfsFileOutStream.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.Closer;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Supports writing to a UFS file directly.
+ */
+@ThreadSafe
+public class UfsFileOutStream extends FileOutStream {
+  /** Used to manage closeable resources. */
+  private final Closer mCloser = Closer.create();
+  private final OutputStream mUfsOutStream;
+
+  /**
+   * Creates a {@link UfsFileOutStream}.
+   *
+   * @param stream the embedded UFS output stream
+   */
+  public UfsFileOutStream(OutputStream stream) {
+    mUfsOutStream = Preconditions.checkNotNull(stream);
+    mCloser.register(mUfsOutStream);
+  }
+
+  @Override
+  public synchronized long getBytesWritten() {
```

Comment:
e.g. 
Fuse.create() creates the out stream
Fuse.write() write to out stream
...
Fuse.getAttr() wants to know how much bytes are written
Fuse.write()
Fuse.getAttr()
Fuse.close()

So I could keep the information on the FUSE side but if the client side already has such information, it would be easier to maintain and get.
Ideally the underlying UFS can directly tell us bytesWritten, but normally the bytesWritten given by UFSes are incorrect (because of buffering related stuff)